### PR TITLE
[Compression app]: add tests

### PR DESCRIPTION
**Pull request by sergii-nosachenko**
## Description

Added tests, improved readme

### Screenshots

Screenshot 2024-01-19 at 12 21 44

### CODE REVIEW

File: tests/createServer.test.js [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+      it('should return a page with a compression select field named "compressionType"', async() => {
+        const { body } = await request();
+
+        expect(body).toMatch(/<select[^>]+name="compressionType"/);
+      });
+
+      it('should return a a page with the list of compression options with values "gzip", "deflate" and "br"', async() => {
+        const { body } = await request();
+
+        const expectedOptions = ['gzip', 'deflate', 'br'];
```
Comment (alexandra-protyanova) [inline tests/createServer.test.js:167]:
```suggestion
        const expectedOptions = Object.keys(compressionTypes)
```

File: tests/createServer.test.js [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+        expectedOptions.forEach(option => {
+          const optionRegex = new RegExp(`<option[^>]+value="${option}"`);
+
+          expect(body).toMatch(optionRegex);
+        });
+      });
+
+      it('should return a a page with a submit button', async() => {
```
Comment (alexandra-protyanova) [inline tests/createServer.test.js:176]:
```suggestion
      it('should return a page with a submit button', async() => {
```

**Review by alexandra-protyanova — APPROVED** [review]
my like goes here

File: tests/createServer.test.js [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+      let response;
+
+      try {
+        response = await axios.post(`${HOST}/non-existing`);
+      } catch (err) {
+        expect(err.response.status)
+          .toBe(404);
+      }
+
+      expect(response).toBeUndefined();
```
[the remark this answers is not in the document]
Comment (mgrinko) [reply tests/createServer.test.js:100]: And here is one more simpler alternative
```suggestion
      return expect(axios.get(COMPRESS_ENDPOINT))
        .rejects
        .its('response')
        .its('status')
        .toBe(400);
```

**Review by mgrinko — CHANGES_REQUESTED** [review]

File: tests/createServer.test.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            })
+              .then(
+                async(res) => {
+                  const compressedData = res.data;
+                  const uncompressedData = await decompress(compressedData);
+
+                  expect(uncompressedData.toString())
+                    .toBe(content);
+                }
+              );
```
Comment (mgrinko) [inline tests/createServer.test.js:217]:
```suggestion
              .then(res => decompress(res.data))
              .then(uncompressedData => {
                expect(uncompressedData.toString())
                  .toBe(content);
              });
```

**Review by mgrinko — APPROVED** [review]

**Pull request merged**